Sources/DownloadManager/ViewModel/DownloadManager.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  DownloadManager.swift
+//
+//
+//  Created by Abhishek Pandey on 29/08/23.
+//
+
+import Foundation
+import SwiftUI
+
+public class DownloadManager: ObservableObject {
+    
+    //MARK: - properties
+    var playerManager = PlayerManager()
+    private var downloadManager = DownloadTask()
+    @Published public var taskResult: TaskResult = .progress(0.0)
+    
+    //MARK: Initializer
+    public init() {
+        //configNotification(isRequire: true)
+        downloadManager.downloadAudioCallback = { [self] result in
+            DispatchQueue.main.async {
+                self.taskResult = result
+            }
+        }
+    }
+    
+    //MARK: downloadAudio
+    public func downloadMedia(with url: String, successTitle: String = "", successSubtitle: String = "") {
+        let mediaName = url.components(separatedBy: "/").last ?? ""
+        downloadManager.successNotificationTitle = successTitle.isEmpty ? notificationTitle : successTitle
+        downloadManager.successNotifcationSubtitle  = successSubtitle.isEmpty ? notificationDescription(mediaName) : successSubtitle
+        downloadManager.downloadMedia(with: url)
+    }
+    
+    //MARK: cancelDownload
+    public func cancelDownload(with url : String, notificationTitle: String = "", notificationSubtitle: String = "") {
+        let mediaName = url.components(separatedBy: "/").last ?? ""
+        downloadManager.cancelNotificationTitle = notificationTitle.isEmpty ? cancelMediaTitle  : notificationTitle
+        downloadManager.cancelNotificationSubtitle = notificationSubtitle.isEmpty ? cancelMediaSubtitle(mediaName) : notificationSubtitle
+        downloadManager.cancelMedia()
+    }
+    
+    // MARK: - openAudioPlayer
+    public func openAudioPlayer(with url: String) -> some View {
+        playerManager.deinitPlayer()
+        return AudioPlayerView(mediaUrl: url)
+    }
+    
+    public func openVideoPlayer(with url: String) {
+        playerManager.deinitPlayer()
+        playerManager.playVideo(url: url)
+    }
+    
+    //MARK: - isMediaExistInDir
+    // check media is already download or not
+    public func isMediaExistInDir(with url : String) -> Bool {
+        if let _ =  playerManager.getMeidaPath(of: url) {
```

Comment:
Shouldn't playerManager.getMeidaPath(of: url) != nil be the preferred approach here ? Also there is a namingConventionFault !!